radicle-cli/src/commands/remote/add.rs

```diff
@@ -0,0 +1,56 @@
+use radicle::node::tracking::store::Config;
+use radicle::{git::Url, node::TRACKING_DB_FILE, prelude::Id, Profile};
+use radicle_crypto::PublicKey;
+
+use crate::{git, terminal as term};
+
+/// core command to run the `rad remote add ...` subcommand.
+pub fn run(
+    repository: &git::Repository,
+    profile: &Profile,
+    pubkey: &PublicKey,
+    name: &Option<String>,
+    id: Id,
+) -> anyhow::Result<()> {
+    let name = if let Some(name) = name {
+        name.to_owned()
+    } else if let Some(name) = lookup_alias(profile, pubkey)? {
+        name
+    } else {
+        anyhow::bail!("an alias need to be specified");
+    };
+
+    if git::rad_has_remote(repository, &name)? {
+        anyhow::bail!("remote with did `{pubkey}` already present");
+    }
+    let url = Url::from(id).with_namespace(*pubkey);
+    let (name, url) = add_new_remote(repository, &name, &url)?;
+
+    term::success!("Remote {name} added with {url}");
+    Ok(())
+}
+
+/// from a node pubkey try to get the alias of the node
+fn lookup_alias(profile: &Profile, pubkey: &PublicKey) -> anyhow::Result<Option<String>> {
+    let path = profile.home.node().join(TRACKING_DB_FILE);
+    let storage = Config::reader(path)?;
+    let Some(node) = storage.node_policy(pubkey)? else {
+        return Ok(None);
+    };
+    Ok(node.alias)
```

Comment:
```suggestion
    Ok(storage.node_policy(pubkey)?.map(|node| node.alias))
```

Reply:
Mh this does not compile, this return a `Result<Option<Option<String>>>`, and I think this makes the code hard to read anyway?

What is wrong with my original solution?

Reply:
Ah, `alias` is `Option`! You can use `and_then` instead of `map`.

I find the `let .. else` quite verbose and the `and_then` and `map` functions exist exactly for these patterns.